### ISSUE TITLE
Сompilation error due to static_assert()

### ISSUE THREAD

**Issue by max-509** [vincentlaucsb/csv-parser#163]
Hello!

First, thank you for excellent library. The speed of operation is only 2.5 times less than the speed in RAM on 800Mb memory.

I want to automatically load your library in a cmake project as an external module. But the automatic build crashes at compile time.

Compilation with c++11 does not work because static_assert(bool) is only after c++17, you need static_assert(bool, message) instead static_assert(bool). In file `common.hpp:170`:

```c++
static_assert(ParseFlags::DELIMITER < ParseFlags::NEWLINE);

/** Optimizations for reducing branching in parsing loop
*
*  Idea: The meaning of all non-quote characters changes depending
*  on whether or not the parser is in a quote-escaped mode (0 or 1)
*/
static_assert(quote_escape_flag(ParseFlags::NOT_SPECIAL, false) == ParseFlags::NOT_SPECIAL);
static_assert(quote_escape_flag(ParseFlags::QUOTE, false) == ParseFlags::QUOTE);
static_assert(quote_escape_flag(ParseFlags::DELIMITER, false) == ParseFlags::DELIMITER);
static_assert(quote_escape_flag(ParseFlags::NEWLINE, false) == ParseFlags::NEWLINE);

static_assert(quote_escape_flag(ParseFlags::NOT_SPECIAL, true) == ParseFlags::NOT_SPECIAL);
static_assert(quote_escape_flag(ParseFlags::QUOTE, true) == ParseFlags::QUOTE_ESCAPE_QUOTE);
static_assert(quote_escape_flag(ParseFlags::DELIMITER, true) == ParseFlags::NOT_SPECIAL);
static_assert(quote_escape_flag(ParseFlags::NEWLINE, true) == ParseFlags::NOT_SPECIAL);
```

Thank for help.

**Comment by vincentlaucsb** [comment]
Thank you for pointing out this compatibility issue.

**Comment by vincentlaucsb** [comment]
Fixed by #179
